src/threading.ts

```diff
@@ -247,11 +270,18 @@ export class BytecodeStackFrame implements IStackFrame {
 
     // Run until we get the signal to return to the thread loop.
     while (!this.returnToThreadLoop) {
-      var op = code.readUInt8(this.pc);
-      if (!RELEASE && logging.log_level === logging.VTRACE) {
-        vtrace(`  ${this.pc} ${annotateOpcode(op, this, code, this.pc)}`);
+      var op = method.getOp(this.pc, code);
```

Comment:
Counting # of opcodes executed seems excessive. I would be more in favor of number of function executions or, if you want to handle long-running loops, basic block executions.